# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/rxuuni/flyrank-F1-week-1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [1]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


**Lane 2 — Refresh / Content Opportunity Scoring**

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


**Which pages should be reviewed first for content refresh based on evidence of decline, demand, and opportunity?**

**Review the highest-ranked pages and decide whether to refresh, expand, protect, prune, or monitor them.**

**A false positive could waste limited review time on a page that does not need intervention, while a false negative could cause an important page to be overlooked.**

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
import os
import pandas as pd

repo_name = "flyrank-F1-week-1"

if not os.path.exists(repo_name):
    !git clone https://github.com/rxuuni/flyrank-F1-week-1.git

os.chdir(os.path.join(repo_name, "work", "notebooks"))

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

declining_pct = (df["trend_direction"] == "down").mean()

print(f"Number of pages: {len(df):,}")
print(f"Pages with declining trend: {declining_pct:.1%}")
print(f"Median impressions (90d): {df['impressions_90d'].median():,.0f}")

Cloning into 'flyrank-F1-week-1'...
remote: Enumerating objects: 133, done.
remote: Counting objects: 100% (133/133), done.
remote: Compressing objects: 100% (89/89), done.
remote: Total 133 (delta 46), reused 92 (delta 28), pack-reused 0 (from 0)
Receiving objects: 100% (133/133), 1.85 MiB | 9.50 MiB/s, done.
Resolving deltas: 100% (46/46), done.
Number of pages: 30,000
Pages with declining trend: 54.2%
Median impressions (90d): 731


**The starter dataset contains 30,000 pages, and 54.2% of them have a declining trend. The median number of impressions over the last 90 days is 731. These numbers suggest that there is a large set of pages to prioritize, while not every page has the same level of demand. This makes a ranking approach useful for identifying which pages should receive attention first.**

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


**The starter dataset contains 30,000 pages, and 54.2% of them have a declining trend. The median number of impressions over the last 90 days is 731. These numbers suggest that there is a large set of pages to prioritize, while not every page has the same level of demand. This makes a ranking approach useful for identifying which pages should receive attention first.**

## Self-check

Before you submit, confirm each line honestly:

- [✅ ] Every section above is filled — markdown thinking AND the code that backs it
- [✅ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [✅ ] No client names, URLs, or private queries anywhere
- [ ✅] My claims use careful words: observed, measured, directional, decision-support
- [ ✅] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.